homeassistant/components/manual/alarm_control_panel.py

```diff
@@ -362,10 +368,12 @@ def _update_state(self, state):
                 self.async_update_ha_state,
                 self._state_ts + pending_time + trigger_time,
             )
-        elif state in SUPPORTED_PENDING_STATES and pending_time:
-            track_point_in_time(
-                self._hass, self.async_update_ha_state, self._state_ts + pending_time
-            )
+        elif state in SUPPORTED_ARMING_STATES:
+            arming_time = self._arming_time(state)
+            if arming_time:
+                track_point_in_time(
+                    self._hass, self.async_update_ha_state, self._state_ts + arming_time
```

Comment:
We need to create out own callback that calls `self.async_write_ha_state`. It's not good to use `self.async_update_ha_state` directly as the callback as we're passing the wrong argument to it from the event listener.

Please fix this for all the cases where we use `track_point_in_time` here.

Reply:
I just commited a fix for this, I tested it locally and it still works and does not give errors, could you check if I implemented it the way you had it in mind?